runtime/vm/CRIUHelpers.cpp

```diff
@@ -263,6 +264,28 @@ criuRestoreInitializeTrace(J9VMThread *currentThread, void *userData, const char
 	return result;
 }
 
+/**
+ * An internal JVM checkpoint hook to disable the shared class cache if the -Xshareclasses:disableOnRestore is present
+ *
+ * @param[in] currentThread vmThread token
+ * @param[in] userData J9InternalHookRecord pointer
+ *
+ * @return BOOLEAN TRUE if no error, otherwise FALSE
+*/
+static BOOLEAN
+criuRestoreDisableSharedClassCache(J9VMThread *currentThread, void *userData, const char **nlsMsgFormat)
+{
+	J9JavaVM *vm = currentThread->javaVM;
+
+	if (NULL != vm->checkpointState.restoreArgsList) {
+		if (FIND_AND_CONSUME_ARG(vm->checkpointState.restoreArgsList, EXACT_MATCH, VMOPT_XSHARECLASSES_DISABLEONRESTORE, NULL) >= 0) {
```

Comment:
There should be a NULL check of vm->sharedClassConfig. If it is NULL, the shared class is already disabled.

Reply:
Added a check in [86ce34e](https://github.com/eclipse-openj9/openj9/commit/86ce34e9e82cb6489368e41b35512d4681859702)